compiler/util/src/org/jetbrains/kotlin/utils/SmartHashMap.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.utils
+
+import java.util.IdentityHashMap
+
+const val ARRAY_UNTIL_SIZE = 10
```

Comment:
I'd probably make it private or even put within a companion for `SmartHashMap` to avoid its appearing in other scopes

Reply:
Done